src/sonic-host-services/scripts/hostcfgd

```diff
@@ -987,16 +994,32 @@ class HostConfigDaemon:
         # Initialize AAACfg
         self.hostname_cache=""
         self.aaacfg = AaaCfg()
+
+        # Initialize cache
+        self.cache = {}
         
 
     def load(self):
+        features = self.config_db.get_table('FEATURE')
+        self.feature_handler.update_all_features_config(features)
```

Comment:
I think it's better to read the data from all the tables first and then start calling load().

If not, the probability of acting upon redundant notifications increases. 
Eg: Listen at t=0, Notif for AAA|xx at t=2, Notif for AAA|xx at t=4, Read the AAA table data at t=10 and act upon it. 

The Notifs at t=2,4 will be cached in the EventBuffer of pubsub and the process will handle those thinking they are latest events causing an unstable state until the latest event is handled.

Reply:
I agree even though we read just after listen(), there is a very slight possibility of this happening but that is the case for SubscriberStateTable as well and this can't be avoided.

Also, i would suggest move the logic of reading the initial data to register_callback() method just after listen(), that is probably the best we can do.

Reply:
The cache should prevent redudant notifications, it will check and see if the data which hostcfgd has loaded in is up to date and will not fire a redundant notification if so. 

I will group all the table loads together so they are maximally in sync.

Reply:
I agree cache can prevent redundant notification but I'm talking about the scenario where the cache has an update that is latest when compared to the notifications saved in the event buffer

Reply:
Not sure I understand you here. The notification in the buffer is always latest as we pull the latest data from CONFIG_DB for the given key before firing the callback.